Fix 500 error encountered with sendTransfer API

We are seeing a persistent issue trying to send bitcoin tx from LeatherProvider as a request to ordinals explorer seems to throw a 500 resulting in the rpc call to crash

```
inscriptions.query.ts:258 
 GET https://ordinals-explorer.generative.xyz/output/f30ac99…:1 500 (Internal Server Error)
logger.ts:51 Error broadcasting tx TypeError: Cannot read properties of undefined (reading 'url')
    at getInscriptionsByTransactionId (bitcoin-client.ts:136:15)
    at use-check-utxos.ts:62:64
    at Array.map (<anonymous>)
    at checkInscribedUtxosByBestinslot (use-check-utxos.ts:62:54)
    at use-check-utxos.ts:147:40
    at async use-bitcoin-broadcas…ransaction.ts:45:37
    at async onUserApproveSendTransferRequest (rpc-send-transfer-confirmation.tsx:92:5)
```
﻿
﻿
<img width="1037" alt="image" src="https://github.com/leather-wallet/extension/assets/488217/14d370e2-08af-4962-8625-8dc0a2af1cbf">

```
  executeTransfer: async (transaction: Transaction) => {
    try {
      const resp = await window.LeatherProvider?.request("sendTransfer", {
        network: "mainnet",
        recipients: [
          {
            address: transaction.destination_address,
            amount: transaction.amount.toString(),
            from: transaction.sending_address,
          },
        ],
      });
      return { txId: resp?.result.txid || null };
    } catch (err) {
      console.log(err);
      throw err;
    }
  },
```
 
we have a project going live in < 12 hours. would great if someone could shed some light on this issue or point out if we're doing something wrong here.

--- a/src/app/query/bitcoin/bitcoin-client.ts
+++ b/src/app/query/bitcoin/bitcoin-client.ts
@@ -54,12 +54,12 @@ interface BestinslotInscription {
   byte_size: number;
 }
 
-export interface BestinslotInscriptionByIdResponse {
+interface BestinslotInscriptionByIdResponse {
   data: BestinslotInscription;
   block_height: number;
 }
 
-export interface BestinslotInscriptionsByTxIdResponse {
+interface BestinslotInscriptionsByTxIdResponse {
   data: { inscription_id: string }[];
   blockHeight: number;
 }

--- a/src/app/query/bitcoin/transaction/use-check-utxos.ts
+++ b/src/app/query/bitcoin/transaction/use-check-utxos.ts
@@ -10,10 +10,7 @@ import { useAnalytics } from '@app/common/hooks/analytics/use-analytics';
 import { useBitcoinClient } from '@app/store/common/api-clients.hooks';
 import { useCurrentNetworkState } from '@app/store/networks/networks.hooks';
 
-import type {
-  BestinslotInscriptionByIdResponse,
-  BestinslotInscriptionsByTxIdResponse,
-} from '../bitcoin-client';
+import type { BitcoinClient } from '../bitcoin-client';
 import { getNumberOfInscriptionOnUtxoUsingOrdinalsCom } from '../ordinals/inscriptions.query';
 
 class PreventTransactionError extends Error {
@@ -45,27 +42,29 @@ export function filterOutIntentionalUtxoSpend({
 interface CheckInscribedUtxosByBestinslotArgs {
   inputs: btc.TransactionInput[];
   txids: string[];
-  getInscriptionsByTransactionId(id: string): Promise<BestinslotInscriptionsByTxIdResponse>;
-  getInscriptionById(id: string): Promise<BestinslotInscriptionByIdResponse>;
+  client: BitcoinClient;
 }
 
 async function checkInscribedUtxosByBestinslot({
   inputs,
   txids,
-  getInscriptionsByTransactionId,
-  getInscriptionById,
+  client,
 }: CheckInscribedUtxosByBestinslotArgs): Promise<boolean> {
   /**
    * @description Get the list of inscriptions moving on a transaction
    * @see https://docs.bestinslot.xyz/reference/api-reference/ordinals-and-brc-20-and-bitmap-v3-api-mainnet+testnet/inscriptions
    */
-  const inscriptionIdsList = await Promise.all(txids.map(id => getInscriptionsByTransactionId(id)));
+  const inscriptionIdsList = await Promise.all(
+    txids.map(id => client.BestinslotApi.getInscriptionsByTransactionId(id))
+  );
 
   const inscriptionIds = inscriptionIdsList.flatMap(inscription =>
     inscription.data.map(data => data.inscription_id)
   );
 
-  const inscriptionsList = await Promise.all(inscriptionIds.map(id => getInscriptionById(id)));
+  const inscriptionsList = await Promise.all(
+    inscriptionIds.map(id => client.BestinslotApi.getInscriptionById(id))
+  );
 
   const hasInscribedUtxos = inscriptionsList.some(resp => {
     return inputs.some(input => {
@@ -147,8 +146,7 @@ export function useCheckInscribedUtxos(blockTxAction?: () => void) {
         const hasInscribedUtxo = await checkInscribedUtxosByBestinslot({
           inputs,
           txids,
-          getInscriptionsByTransactionId: client.BestinslotApi.getInscriptionsByTransactionId,
-          getInscriptionById: client.BestinslotApi.getInscriptionById,
+          client,
         });
 
         if (hasInscribedUtxo) {

--- a/src/shared/constants.ts
+++ b/src/shared/constants.ts
@@ -76,7 +76,7 @@ export interface NetworkConfiguration {
 }
 
 export const BESTINSLOT_API_BASE_URL_MAINNET = 'https://leatherapi.bestinslot.xyz/v3';
-export const BESTINSLOT_API_BASE_URL_TESTNET = 'https://testnet.api.bestinslot.xyz/v3';
+export const BESTINSLOT_API_BASE_URL_TESTNET = 'https://leatherapi_testnet.bestinslot.xyz/v3';
 
 export const HIRO_API_BASE_URL_MAINNET = 'https://api.hiro.so';
 export const HIRO_API_BASE_URL_TESTNET = 'https://api.testnet.hiro.so';

--- a/tests/specs/send/send-btc.spec.ts
+++ b/tests/specs/send/send-btc.spec.ts
@@ -100,9 +100,65 @@ test.describe('send btc', () => {
 
       await sendPage.clickInfoCardButton();
 
-      const isErrorPageVisible = await sendPage.broadcastErrorTitle.isVisible();
+      await test.expect(sendPage.broadcastErrorTitle).toBeVisible();
+    });
 
-      test.expect(isErrorPageVisible).toBeTruthy();
+    test('that fallbacks to other api provider if main fails', async ({ sendPage }) => {
+      let output = '';
+      let id = '';
+      let index = '';
+
+      await sendPage.page.route('**/ordinals-explorer.generative.xyz/**', async route => {
+        return route.fulfill({
+          status: 500,
+          contentType: 'text/html',
+          body: mockOrdinalsComApiHtmlResponse,
+        });
+      });
+
+      sendPage.page.on('request', async request => {
+        if (request.url().includes('ordinals-explorer.generative.xyz')) {
+          const url = request.url();
+          output = url.split('/').pop() || '';
+          id = output.split(':')[0];
+          index = output.split(':')[1];
+        }
+      });
+
+      await sendPage.page.route(
+        '**/leatherapi.bestinslot.xyz/v3/inscription/in_transaction**',
+        async route => {
+          return route.fulfill({
+            status: 200,
+            contentType: 'application/json',
+            body: JSON.stringify({
+              data: [{ txid: id, output, index, satpoint: output }],
+            }),
+          });
+        }
+      );
+
+      await sendPage.page.route(
+        '**/leatherapi.bestinslot.xyz/v3/inscription/single_info_id**',
+        async route => {
+          return route.fulfill({
+            status: 200,
+            contentType: 'application/json',
+            body: JSON.stringify({
+              data: { txid: id, output, index, satpoint: output },
+            }),
+          });
+        }
+      );
+      await sendPage.amountInput.fill('0.00006');
+      await sendPage.recipientInput.fill(TEST_TESTNET_ACCOUNT_2_BTC_ADDRESS);
+
+      await sendPage.previewSendTxButton.click();
+      await sendPage.feesListItem.filter({ hasText: BtcFeeType.High }).click();
+
+      await sendPage.clickInfoCardButton();
+
+      await test.expect(sendPage.broadcastErrorTitle).toBeVisible();
     });
   });
 });

